app-cpp/app355.cpp

```diff
@@ -393,6 +324,81 @@ void SatOrbit::OnSortOrbitalVectorMulti(orbit_iterator& inBegin, orbit_iterator&
     return inLHS.GetTLE().GetMeanMotion() < inRHS.GetTLE().GetMeanMotion();
 }
 
+void SatOrbit::SortOrbitalVectorMulti(std::vector<OrbitalData>& ioOrbitalVector)
+{
+    const std::size_t orbitVectorSize = ioOrbitalVector.size();
+
+        // calculate the number of TLEs per thread
+        const std::size_t orbitPerThread = orbitVectorSize / mNumThreads;
+        // Note: no need for mutex as threads do not contend for shared elements during sort
+        IteratorPairVector subListVector{};
+
+        {
```

Comment:
You need a comment here:
Something saying: block scope that has multiple threads quicksort segments of the vector